### PR TITLE
Checkin document pages

### PR DESCRIPTION
**Pull request by yiheng**
## What changes were proposed in this pull request?
This PR checkin several document pages into the doc folder

This document is for
* SpatialDilatedConvolution
* CMinTable
* RReLU
* VolumetricMaxPooling
* Add
* Select
* SplitTable
* TimeDistributed
* DistriOptimizer

## How was this patch tested?
Unit test

### CODE REVIEW

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/RReLU.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+```python
+layer = RReLU(lower, upper, inPlace)
+```
+
+Applies the randomized leaky rectified linear unit (RReLU) element-wise to the input Tensor,
+thus outputting a Tensor of the same dimension. Informally the RReLU is also known as 'insanity' layer.
+
+RReLU is defined as: f(x) = max(0,x) + a * min(0, x) where a ~ U(l, u).
+
+In training mode negative inputs are multiplied by a factor a drawn from a uniform random
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/RReLU.md]: a typo here? "by a factor drawn from"?

File: docs/docs/APIdocs/layers/nn/Pooling Layers/VolumetricMaxPooling.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+**Python:**
+```python
+layer = VolumetricMaxPooling(
+  kernelT, kernelW, kernelH,
+  strideT, strideW, strideH,
+  paddingT, paddingW, paddingH
+)
+```
+
+Applies 3D max-pooling operation in kT x kW x kH regions by step size dT x dW x dH steps.
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/nn/Pooling Layers/VolumetricMaxPooling.md]: a little confuse about this expression.  do you mean "by step size dT x dW x dH" without the last "steps"?

File: docs/docs/APIdocs/layers/nn/Utilities/SplitTable.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+```scala
+val layer = SplitTable[T](dim)
+```
+**Python:**
+```python
+layer = SplitTable(dim)
+```
+
+SplitTable takes a Tensor as input and outputs several tables,
+splitting the Tensor along the specified dimension `dimension`. Please not
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/nn/Utilities/SplitTable.md]: "please note"

File: docs/docs/APIdocs/layers/nn/Utilities/SplitTable.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+the dimension start from 1.
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/nn/Utilities/SplitTable.md]: "starts"

File: docs/docs/APIdocs/layers/nn/Utilities/SplitTable.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+The input to this layer is expected to be a tensor, or a batch of tensors;
+when using mini-batch, a batch of sample tensors will be passed to the layer and
+the user need to specify the number of dimensions of each sample tensor in a
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/nn/Utilities/SplitTable.md]: "need"

Comment (yangw1234) [inline docs/docs/APIdocs/layers/nn/Utilities/SplitTable.md]: "needs"

File: docs/docs/APIdocs/layers/optim/DistriOptimizer.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+* **data**
+
+Your training data. As we train models on Spark, one of
+the most common distributed data structure is RDD.Of course
+you can use DataFrame, Please check the BigDL pipeline example.
+
+The element in the RDD is Sample, which is actually a sequence of
+Tensors. You need to convert your data record(image, audio, text)
+to Tensors before you feed them into Optimizer. We also provides
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/optim/DistriOptimizer.md]: "provide"

File: docs/docs/APIdocs/layers/optim/DistriOptimizer.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+many utilities to do it.
+
+* **loss function**
+
+In supervised machine learning, loss function compare the output of
+the model with the ground truth(the labels of the training data). It
+outputs a loss value to measure how good the model is(the lower the
+better). It also provide a gradient to indicate how to tune the model.
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/optim/DistriOptimizer.md]: "provides"

File: docs/docs/APIdocs/layers/optim/DistriOptimizer.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+In BigDL, all loss functions are subclass of Criterion.
+
+* **batch size**
+
+Training is an iterative process. In each iteration, only a batch of data
+is used for training the model. You need to specify the batch size. Please note, 
+the batch size should be divided by the total cores number.
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/optim/DistriOptimizer.md]: "should be divisible"?

File: docs/docs/APIdocs/layers/optim/DistriOptimizer.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+
+You can see the model is trained.
+
+### Define when to end the training
+You need define when to end the training. It can be several iterations, or how many round
+data you want to process, a.k.a epoch.
+
+**scala**
+```scala
+// The define endWhen in scala is 100 iterations
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/optim/DistriOptimizer.md]: do you mean "the default endWhen"?

File: docs/docs/APIdocs/layers/optim/DistriOptimizer.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+```
+
+**python**
+```
+# Python need to define in the constructor
+optimizer = Optimizer(model, train_data, MSECriterion(), MaxIteration(100), 4)
+```
+
+### Change the optimization algorithm
+Gradient based optimization algorithm is the most popular algorithms to train the neural
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/optim/DistriOptimizer.md]: "Gradient based optimization algorithms are the most popular algorithms"?

File: docs/docs/APIdocs/layers/optim/DistriOptimizer.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+network model. The most famous one is SGD. SGD has many variants, adagrad, adam, etc.
+
+**scala**
+```scala
+// The define is SGD
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/optim/DistriOptimizer.md]: "default"?

File: docs/docs/APIdocs/layers/optim/DistriOptimizer.md [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+optimizer.setOptimMethod(new Adam())  // Change to 10 epoch
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/optim/DistriOptimizer.md]: "change to adam"?

File: docs/docs/APIdocs/layers/optim/DistriOptimizer.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+
+**python**
+```
+# Python need to define in the constructor
+optimizer = Optimizer(model, train_data, MSECriterion(), MaxIteration(100), 4, optim_method = Adam())
+```
+
+### Validate your model in training
+Sometimes, people want to evaluate the model being trained with a seperated dataset. When model
+perform well on train dataset, but bad on validation dataset, we call the model is overfit or
```
Comment (yangw1234) [inline docs/docs/APIdocs/layers/optim/DistriOptimizer.md]: "performs"?
